analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CastShouldNotBeDuplicated.CSharp11.cs

```diff
@@ -7,9 +7,22 @@ void ListPattern()
     {
         object[] numbers = { 1, 2, 3 };
 
-        if (numbers is [double something, 3, 3])    // FN
+        if (numbers is [Fruit fruit, 3, 3])    // Secondary
+//                      ^^^^^^^^^^^
         {
-            var ff1 = (double)something;            // FN
+            var ff1 = (Fruit)fruit;            // Noncompliant
+        }
+
+        if (numbers is [double number, 3, 3])
+        {
+            var ff2 = (double)number;          // Compliant (casting to a ValueType)
```

Comment:
The rule is explicitly excluding all the cases witch casts to a ValueType.
Until further modifications, it will be considered compliant.

Reply:
Could you please update the reproducer in the issue to not have a ValueType?